### PR TITLE
Allow customization on data and error

### PR DESCRIPTION
**Pull request by bruno-delfino1995**
Now we can use a reducer for customizing the state on data and error properties. Before this, if you wanted to have a paginated resource, you'd need to capture the async actions and append/process the data outside of the reducer given by `createR`.

By accepting a second optional parameter as a reducer, `createR` streams the actions to it and uses the returned state. Therefore, to handle paginated resources, you would pass a reducer which joins incoming with current data.

```javascript
const concatReducer = (state, action) => {
  const { type, payload } = action

  switch (type) {
    case STATUS.PENDING: return { ...state, error: null }
    case STATUS.FULFILLED:
      return {
        error: null,
        data: [...state.data, ...payload.elements],
      }
    case STATUS.REJECTED: return { ...state, error: payload }
    default: return state
  }
}

const reducer = createR('FETCH_PRODUCTS', concatReducer)
```

Through the `concatReducer`, we're able to join what we had with the incoming payload, hence, handling continuous pagination.

### CODE REVIEW

File: src/createR.js [hunk 1]
```diff
@@ -4,40 +4,87 @@ const initialState = {
...
   loading: false,
   error: null,
-  data: null
+  data: null,
 }
 
-const createR = name => (state = initialState, action) => {
-  const { type, payload, meta } = action
+const defaultReducer = (state, action) => {
+  const { type, payload } = action
```
Comment (xflavioribeiro) [inline src/createR.js]: Could not use unstructuring of these elements as function parameters?
`cosnt defaultReducer  = (state, { type, payload }) => ...`

File: src/createR.js [hunk 2]
```diff
@@ -4,40 +4,87 @@ const initialState = {
...
+        error: payload,
+        data: null,
+      }
+    default:
+      return state
+  }
+}
 
-  switch (meta.status) {
+const trackingReducer = (state, action) => {
```
Comment (xflavioribeiro) [inline src/createR.js]: Here equal above

File: src/createR.js [hunk 3]
```diff
@@ -4,40 +4,87 @@ const initialState = {
...
         loading: false,
-        error: payload,
-        data: null
       }
     default:
       return state
   }
 }
 
+const guard = name => (action) => {
```
Comment (xflavioribeiro) [inline src/createR.js]: Could not use unstructuring of these elements as function parameters?
`cosnt defaultReducer  = ({ type, meta }) => ...`

File: src/createR.js [hunk 4]
```diff
@@ -4,40 +4,87 @@ const initialState = {
...
+  const statuses = Object.values(STATUS)
+
+  if (!meta || (meta && meta.name !== name)) return false
+  if (!type || !type.startsWith(prefixAT(name))) return false
+  if (!statuses.includes(meta.status)) return false
+
+  return true
+}
+
+const createR = (name, resultReducer = defaultReducer) => (state = initialState, action) => {
```
Comment (xflavioribeiro) [inline src/createR.js]: Could not use unstructuring of these elements as function parameters?
`const createR  = (name, resultReducer = defaultReducer) => (state = initialState, { meta }) => ...`

Comment (bruno-delfino1995) [reply src/createR.js]: I can't use destructuring on this because I need the full action right below for the guard function.

**Pull request merged**